### PR TITLE
减少或者消除 puslar使用过程中的报错和提示

### PR DESCRIPTION
**Pull request by xingkongwuhen**
减少或者消除 puslar使用过程中的报错和提示

### CODE REVIEW

**Comment by IggsGrey** [comment]
To speed things up, above means: "To reduce or eliminate errors and warnings during the use of Puslar"

**Comment by xingkongwuhen** [comment]
修复了再Pulsar编辑器使用的时候出现的包含但不限于configArray在内的几个问题，使得Pulsar上可以正常使用这个FTP插件

**Comment by IggsGrey** [comment]
Correct, basically I don't think it's even good practice to write code in that way. Particularly let and const should be used instead of var as most "experts" say. However given the existing code structure it may not be worth the effort to refactor the entire code base.

So let's hope this one gets merged soon, otherwise I propose that you create a fork of this? Given that you have the time to maintain.

File: lib/ftp-remote-edit.js [hunk 1]
```diff
@@ -1199,7 +1198,8 @@ class FtpRemoteEdit {
     }
 
     let initialPath, initialName, initialType, ref;
```
Comment (cschindl) [inline lib/ftp-remote-edit.js:1200]: Use let instead of var

let entry, initialPath, initialName, initialType, ref;

File: lib/helper/queue.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Queue {
 
   addFile(file) {
     const self = this;
-
+    console.log(file);
```
Comment (cschindl) [inline lib/helper/queue.js:24]: remove debug console output

File: lib/helper/storage.js [hunk 3]
```diff
@@ -53,7 +53,9 @@ class Storage {
         configText = decrypt(self.password, configText);
 
         try {
+          var configArray = [];
```
Comment (cschindl) [inline lib/helper/storage.js:56]: configArray is an object so it should be declared as object and not as an array

let configArray = self.migrate(JSON.parse(cleanJsonString(configText)));

File: lib/helper/storage.js [hunk 4]
```diff
@@ -396,7 +398,7 @@ class Storage {
       self.backup(atom.packages.getActivePackage('ftp-remote-edit').metadata.version);
 
       // migrate
-      let newConfigArray = {};
+      let newConfigArray = [];
```
Comment (cschindl) [inline lib/helper/storage.js:401]: newConfigArray is an object so it should be declared as object and not as an array

let newConfigArray = {};

File: lib/helper/storage.js [hunk 5]
```diff
@@ -53,7 +53,9 @@ class Storage {
         configText = decrypt(self.password, configText);
 
         try {
+          var configArray = [];
           configArray = self.migrate(JSON.parse(cleanJsonString(configText)));
+          console.log(configArray);
```
Comment (cschindl) [inline lib/helper/storage.js:58]: remove debug console output

File: lib/ftp-remote-edit.js [hunk 6]
```diff
@@ -266,7 +266,7 @@ class FtpRemoteEdit {
       if (selected.length === 0) return;
 
       let root = selected.view().getRoot();
-      let localPath = normalize(root.getLocalPath());
+      var localPath = normalize(root.getLocalPath());
```
Comment (cschindl) [inline lib/ftp-remote-edit.js:269]: Use let instead of var

let localPath = normalize(root.getLocalPath());

File: lib/ftp-remote-edit.js [hunk 7]
```diff
@@ -1925,7 +1925,7 @@ class FtpRemoteEdit {
 
       self.finderView.on('ftp-remote-edit-finder:open', (item) => {
         let relativePath = item.relativePath;
-        let localPath = normalize(self.finderView.root.getLocalPath() + relativePath, Path.sep);
+        var localPath = normalize(self.finderView.root.getLocalPath() + relativePath, Path.sep);
```
Comment (cschindl) [inline lib/ftp-remote-edit.js:1928]: Use let instead of var

let localPath = normalize(self.finderView.root.getLocalPath() + relativePath, Path.sep);

**Review by cschindl — CHANGES_REQUESTED** [review]
Hi xingkongwuhen,

thank you for your support. In my opinion, there are still a few small things that you need to adjust.

File: lib/ftp-remote-edit.js (hunk 7 above)
Comment (xingkongwuhen) [reply lib/ftp-remote-edit.js:1928]: Thank you for your suggestions and corrections. I will adjust a version as soon as possible. Thank you again

**Review by xingkongwuhen — COMMENTED** [review]
我认为他是有效的

**Comment by cschindl** [comment]
Thanks for the changes. I have merged everything and now try to publish a new version. Unfortunately it is not as easy as it was under atom. (https://github.com/orgs/pulsar-edit/discussions/91)

**Pull request merged**